ethtxmanager/ethtxmanager.go

```diff
@@ -254,6 +280,8 @@ func (c *Client) monitorTxs(ctx context.Context) error {
 
 	log.Infof("found %v monitored tx to process", len(mTxs))
 
+	nonceMap := newNonceStore()
```

Comment:
Do you think it should be created per Client struct rather than passing it as a function argument

Reply:
That is something to discuss. To me it is probably enough to create this `nonceStore` per monitoring cycle.
If we are to store the nonce map in the `Client` struct itself, then, I am not sure when is the right time to clean it? It might grow, and take memory space unnecessarily.